ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Ouroboros.Network.Protocol.LocalStateQuery.Client (
+      -- * Protocol type for the client
+      -- | The protocol states from the point of view of the client.
+      LocalStateQueryClient(..)
+    , ClientStIdle(..)
+    , ClientStAcquiring(..)
+    , ClientStAcquired(..)
+    , ClientStQuerying(..)
+
+      -- * Execution as a typed protocol
+    , localStateQueryClientPeer
+    ) where
+
+import           Network.TypedProtocol.Core
+
+import           Ouroboros.Network.Block (Point)
+import           Ouroboros.Network.Protocol.LocalStateQuery.Type
+
+
+newtype LocalStateQueryClient block query result m a = LocalStateQueryClient {
```

Comment:
There's no need for the new type wrapper, but let it be this way.  We plan to remove all the top level newtype wrappers from all of the mini-protocols.

Reply:
Ok. I'll leave it there for consistency.